src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -109,14 +117,124 @@ protected Object readResolve() throws ObjectStreamException {
      * <p>It will be false for a node which still has active children, like a step with a running body.
      * It will also be false for something that has finished but is pending child node creation,
      * such as a completed fork branch which is waiting for the join node to be created.
-     * <p>This can only go from true to false and is a shortcut for {@link FlowExecution#isCurrentHead}.
+     * <p>This can only go from true to false.
+     * @deprecated Usually {@link #isActive} is what you want. If you really wanted the original behavior, use {@link FlowExecution#isCurrentHead}.
      */
-    @Exported
+    @Deprecated
     public final boolean isRunning() {
         return getExecution().isCurrentHead(this);
     }
 
     /**
+     * Checks whether a node is still part of the active part of the graph.
+     * Unlike {@link #isRunning}, this behaves intuitively for a {@link BlockStartNode}:
+     * it will be considered active until the {@link BlockEndNode} is added.
+     */
+    @Exported(name="running")
+    public final boolean isActive() {
+        if (this instanceof FlowEndNode) { // cf. JENKINS-26139
+            LOGGER.finer("shortcut: FlowEndNode is never active");
+            return false;
+        }
+        if (this instanceof BlockStartNode) {
+            Map<FlowExecutionOwner, Map<String, Boolean>> startNodesAreClosedByFlow = FlowL.startNodesAreClosedByFlow();
+            LOGGER.log(Level.FINER, "for {0}, startNodesAreClosedByFlow={1}", new Object[] {this, startNodesAreClosedByFlow});
+            Map<String, Boolean> startNodesAreClosed = startNodesAreClosedByFlow.get(exec.getOwner());
+            if (startNodesAreClosed != null) {
+                Boolean closed = startNodesAreClosed.get(id);
+                if (closed != null) { // quick version
+                    LOGGER.log(Level.FINER, "quick closed={0}", closed);
+                    return !closed;
+                } else {
+                    LOGGER.log(Level.FINER, "no record of {0} in {1}, presumably GraphListener not working", new Object[] {this, exec});
```

Comment:
Worth mentioning that this will detect cases where GraphListeners as extensions can't be invoked for all FlowNodes.

```java
Method m = hudson.util.ReflectionUtils.findMethod(CpsFlowExecution.class, "getListenersToRun", null);
            if (m != null) {
                return true;
            }
            return true;
```